### PR TITLE
[Core] Quantum Painter - LVGL Integration

### PR DESCRIPTION
**Pull request by Jpe230**
## Description

This PR aims to integrate  [LVGL (Light and Versatile Embedded Graphics Library)](https://lvgl.io/) into QMK by utilizing Quantum Painter's APIs and drivers.

This PR includes the following:
- Addition of LVGL v8.2 as a submodule.
- Hooking LVGL internal ticks into quantum's main loop by using deferred executions.
- Public API to attach/detach a QP display.
- Enable the use of a `lv_conf.h` user defined file to enable/disable features, much like the rest of the ChibiOS configs. (Courtesy of @tzarc, thanks Tzarc!)
- Documentation of the feature
- Onekey keymap of the feature.

Worth mentioning that English isn't my first language, so I expect to bounce with maintainers the wording of the documentation....

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/onekey/keymaps/lvgl/config.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* SPI pins */
```
Comment (zvecr) [inline keyboards/handwired/onekey/keymaps/lvgl/config.h:1]:
```suggestion
#pragma once 

/* SPI pins */
```

File: quantum/main.c [hunk 2]
```diff
@@ -63,6 +63,12 @@ int main(void) {
         // Run Quantum Painter animations
         void qp_internal_animation_tick(void);
         qp_internal_animation_tick();
+
```
Comment (zvecr) [inline quantum/main.c:66]: Introducing extra logic here would be nice to avoid and instead introduce something like a `qp_task()`

**Review by zvecr — CHANGES_REQUESTED** [review]
* `data/keyboard-config-templates/chibios/` should be `data/templates/<something>`
* some files are missing licence headers

**Comment by tzarc** [comment]
We'll need changes to the root `Makefile` too, specifically for handling submodules.

I *think* that may need to go to master in order to deal with swapping branches and the like, people are likely going to end up with an `lvgl` directory otherwise, and end up adding it to their master-based branches.

**Comment by tzarc** [comment]
And, of course, we'll need a `qmk/lvgl` mirror.

File: docs/_summary.md [hunk 3]
```diff
@@ -99,6 +99,7 @@
   * Hardware Features
     * Displays
       * [Quantum Painter](quantum_painter.md)
+        * [ Quantum Painter LVGL Integration](quantum_painter_lvgl.md)
```
Comment (tzarc) [inline docs/_summary.md:102]:
```suggestion
        * [Quantum Painter LVGL Integration](quantum_painter_lvgl.md)
```

File: docs/quantum_painter_lvgl.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+QUANTUM_PAINTER_LVGL_INTEGRATION = yes
+```
+To configure the Quantum Painter Display Drivers please read the [Quantum Painter Display Drivers](quantum_painter.md#quantum-painter-drivers) section.
+
+## Quantum Painter LVGL API :id=lvgl-api
+
+### Quantum Painter LVGL Attach :id=lvgl-api-init
+
+```c
+bool qp_lvgl_attach(painter_device_t device)
```
Comment (tzarc) [inline docs/quantum_painter_lvgl.md:26]:
```suggestion
bool qp_lvgl_attach(painter_device_t device);
```

File: docs/quantum_painter_lvgl.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+The `qp_lvgl_attach` function is used to set up LVGL with the supplied display, it requires an already configured display.
```
Comment (tzarc) [inline docs/quantum_painter_lvgl.md:29]:
```suggestion
The `qp_lvgl_attach` function is used to set up LVGL with the supplied display, and requires an already configured display.
```

File: docs/quantum_painter_lvgl.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    display = qp_make_.......;         // Create the display
+    qp_init(display, QP_ROTATION_0);   // Initialise the display
+    
+    if (qp_lvgl_attach(display)) {     // Attach LVGL to the display
+        ...Your code to draw           // Run LVGL specific code to draw
+    }
+}
+```
+To init. the display please read the [Display Initialisation](quantum_painter.md#quantum-painter-api-init) section.
+
```
Comment (tzarc) [inline docs/quantum_painter_lvgl.md:43]:
```suggestion

!> Attaching LVGL to a display means LVGL subsequently "owns" the display. Using standard Quantum Painter drawing operations with the display after LVGL attachment will likely result in display artifacts.
```

File: docs/quantum_painter_lvgl.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+### Quantum Painter LVGL Detach :id=lvgl-api-init
+
+```c
+void qp_lvgl_detach()
+```
+
+The `qp_lvgl_detach` function stops the internal LVGL ticks and releases resources related to it.
+
+## Enabling/Disabling LVGL features :id=lvgl-configuring
+You can overwrite LVGL specific features in your `lv_conf.h` file.
```
Comment (tzarc) [inline docs/quantum_painter_lvgl.md:53]:
```suggestion
## Enabling/Disabling LVGL features :id=lvgl-configuring

You can overwrite LVGL specific features in your `lv_conf.h` file.
```

**Review by tzarc — CHANGES_REQUESTED** [review]
Just some verbiage. LGTM otherwise.

**Comment by tzarc** [comment]
Would also be good if you could incorporate your `jpe230/lvgl_makefile_changes` modifications too. We can backport the appropriate lines to `master` afterwards.

**Review by tzarc — APPROVED** [review]

**Review by KarlK90 — APPROVED** [review]
Tested with ST7789 and RP2040 development board, works nicely. Also rather low impact on the scan rate, which is very welcome.

**Pull request merged**